openmeetings-server/src/site/markdown/LdapAndADS.md

```diff
@@ -0,0 +1,238 @@
+<!-- 
+# Licensed under the Apache License, Version 2.0 (the "License") http://www.apache.org/licenses/LICENSE-2.0 -->
+
+# LDAP and Active Directory Integration
+
+This manual explains the configuration to use an already existing
+(e.g. corporate or university) LDAP/AD Server for user authentication
+into OpenMeetings. Successful integration has also been reported with
+Novell eDirectory.
+
+For installation and setup instructions of an LDAP server itself see
+https://help.ubuntu.com/lts/serverguide/openldap-server.html
+
+## LDAP Configuration in OpenMeetings
+
+To configure access to an LDAP Server, three things are needed:
+
+### 1. Information how to access the LDAP Server for user lookup
+
+-   Obtain the ip address and port number of the LDAP server. For the
+    example configuration in the next section we will use the ip
+    address 10.8.0.10 and the default LDAP port number 389.
+
+-   A single LDAP user account is needed that has read privileges to the
+    user entries that are to be used for login by OpenMeetings. Often
+    these accounts have the organizational unit attribute
+    `ou=system-users`.
+
+-   Obtain the cn and password of that user from the LDAP
+    administrator. For the example configuration we will use
+    `cn=openmeetings` and `password=12345`.
+
+-   Obtain the dc values of the root of the LDAP database. In our
+    example we will use `dc=mydomain,dc=org`.
+
+### 2. An LDAP config file
+
+-   This file needs to be located in the folder
+    `$OM_HOME/webapps/openmeetings/data/conf/` <sup><a id="fnr.1" class="footref" href="#fn.1">1</a></sup>.
+    An example file called `om_ldap.cfg` should come with your
+    OpenMeetings installation.
+
+-   The file needs to contain the access information obtained in the
+    previous section. For our example the file would contain the
+    following:
+
+    ```
+    ldap_conn_host=10.8.0.10
+    ldap_conn_port=389
+    ldap_conn_secure=false
+    
+    # The following lines specify the data of the ldap user with 
+    # read privileges to the user entries according to the LDAP DB.
+    ldap_admin_dn=cn=openmeetings,ou=system-users,dc=mydomain,dc=org
+    ldap_passwd=12345
+    
+    # The following line specifies where to search for users wanting to login 
+    ldap_search_base=ou=users,dc=mydomain,dc=org
+    
+    # The attribute in the LDAP DB which will be matched against the 
+    # given login data ("Username or mail address" field) in OpenMeetings:
+    ldap_search_query=(uid=%s)
+    
+    # the scope of the search might be: OBJECT, ONELEVEL, SUBTREE
+    ldap_search_scope=ONELEVEL
+    
+    # Ldap auth type(NONE, SEARCHANDBIND, SIMPLEBIND)
+    #  When using SIMPLEBIND a simple bind is performed on the LDAP server to check user authentication
+    #  When using NONE, the Ldap server is not used for authentication
+    ldap_auth_type=SIMPLEBIND
```

Comment:
with this `ldap_auth_type` there will be no "search for user"
so these params `ldap_admin_dn, ldap_passwd, ldap_search_base, ldap_search_query, ldap_search_scope` has no effect

Reply:
Your suggestions are very god anfd helpful. How shall we go about it? Shall I change it in my repo and issue a new pull request or can you just add your remarks to the source?

Reply:
Just push to the same branch or your repo
The changes will appear in this PR :)

Reply:
ok. Shall I change and 

> Can I ask you to create new screenshot using upcoming M4 ?



> Can I ask you to create new screenshot using upcoming M4 ?

Tomorrow is the opening of the new semester with all our students and I don't want to break anything by upgrading. Is it possible to install a newer version in some other directory and test it with the same DB as the older version before replacing it?

Reply:
> with this `ldap_auth_type` there will be no "search for user"
> so these params `ldap_admin_dn, ldap_passwd, ldap_search_base, ldap_search_query, ldap_search_scope` has no effect

Does that mean that a system-user access isn't strictly necessary in case of "SIMPLEBIND" and the privileged access is only used in "SEARCHANDBIND"?

Reply:
yes

It is also being used in `Command line admin` for user search and import